lms/static/js/views/message.js

```diff
@@ -4,25 +4,29 @@
         'gettext', 'jquery', 'underscore', 'backbone'
     ], function (gettext, $, _, Backbone) {
 
-        var MessageBannerView = Backbone.View.extend({
+        return Backbone.View.extend({
 
-            initialize: function () {
-                this.template = _.template($('#message_banner-tpl').text());
+            initialize: function (options) {
+                var templateId = _.isUndefined(options.templateId) ? '#message_view-tpl' : options.templateId;
+                this.template = _.template($(templateId).text());
             },
 
             render: function () {
                 if (_.isUndefined(this.message) || _.isNull(this.message)) {
                     this.$el.html('');
                 } else {
-                    this.$el.html(this.template({
+                    var data = {
                         message: this.message
-                    }));
+                    };
+                    _.extend(data, this.icon);
+                    this.$el.html(this.template(data));
                 }
                 return this;
             },
 
-            showMessage: function (message) {
+            showMessage: function (message, icon) {
                 this.message = message;
+                this.icon = _.isUndefined(icon) ? {} : {icon: icon};
```

Comment:
Just set `this.icon = icon` here. And then you can add to the context in `render` like variables are generally added.